[CI] GenerativeIT test failing

**Build Scans:**
- [elasticsearch-periodic #11913 / openjdk21_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/mfljo24bzeptm)
- [elasticsearch-periodic-platform-support #11890 / ubuntu-2404-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/bqxmprsl4sbai)
- [elasticsearch-periodic-platform-support #11890 / almalinux-8-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/6rf7uqrszsm6c)
- [elasticsearch-intake #34826 / part3](https://gradle-enterprise.elastic.co/s/2ti5h7c6f426g)
- [elasticsearch-pull-request #122260 / part-3](https://gradle-enterprise.elastic.co/s/qxofnnn5mqqto)
- [elasticsearch-pull-request #122139 / part-3](https://gradle-enterprise.elastic.co/s/ilcte3qv7piwa)
- [elasticsearch-intake #34808 / part3](https://gradle-enterprise.elastic.co/s/dvrungnvsfvco)
- [elasticsearch-pull-request #122124 / part-3](https://gradle-enterprise.elastic.co/s/pmtanpetmfbpg)
- [elasticsearch-pull-request #122121 / part-3](https://gradle-enterprise.elastic.co/s/jcehlcbckpreg)
- [elasticsearch-pull-request #122099 / part-3-arm](https://gradle-enterprise.elastic.co/s/efg6n6u4ywhhg)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT.test" -Dtests.seed=8E056BC34342ADE9 -Dtests.locale=ar-MR -Dtests.timezone=Europe/Bucharest -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: from languages_lookup,messa* | keep `language_code` | mv_expand `language_code` | eval  `language_code` = language_code + language_code, `language_code` = -92733565570034336, ginUJgbg = true, GKlZDeqJJn = 2107629793 | SAMPLE 0.29984937331915007
error: Expecting [3] columns, got [4]
```

**Issue Reasons:**
- [main] 39 failures in test test (12.1% fail rate in 321 executions)
- [main] 26 failures in step part-3 (12.7% fail rate in 204 executions)
- [main] 7 failures in step part3 (15.6% fail rate in 45 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (66.7% fail rate in 3 executions)
- [main] 26 failures in pipeline elasticsearch-pull-request (13.1% fail rate in 198 executions)
- [main] 7 failures in pipeline elasticsearch-intake (15.6% fail rate in 45 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

@luigidellaquila I think this latest failure has to do exactly with what your PR fixed https://github.com/elastic/elasticsearch/pull/142638, only that this run didn't have your code.

For your reference, this is the query

```
SET unmapped_fields="nullify";from servi*,distances
| sort owner
| where distance == 52 OR distance == 92 AND distance != -12 OR  NOT distance != -95 AND  NOT unmapped_field_foo IS NOT NULL
```

and this is the failure:

```
java.lang.AssertionError: query: SET unmapped_fields="nullify";from servi*,distances \| sort owner \| where distance == 52 OR distance == 92 AND distance != -12 OR  NOT distance != -95 AND  NOT unmapped_field_foo IS NOT NULL |  
error: Expecting [4] columns, got [5] |  
at __randomizedtesting.SeedInfo.seed([8E7675164E5B354B:6224ACCE0A758B3]:0) |  
at org.junit.Assert.fail(Assert.java:89) |  
at org.elasticsearch.xpack.esql.qa.rest.generative.GenerativeRestTest.checkResults(GenerativeRestTest.java:291) |  
at org.elasticsearch.xpack.esql.qa.rest.generative.GenerativeRestTest$1.run(GenerativeRestTest.java:199) |  
at org.elasticsearch.xpack.esql.generator.EsqlQueryGenerator.generatePipeline(EsqlQueryGenerator.java:209) |  
at org.elasticsearch.xpack.esql.qa.rest.generative.GenerativeRestTest.test(GenerativeRestTest.java:239)
```

The query as is generates 5 columns, if I remove the `where` it's generating 4 (the `unmapped_field_foo` is the additional one).
Can you confirm, please?

```
Found 2 problems\nline 1:368: Column [client_ip] has conflicting data types in FORK branches: [NULL] and [KEYWORD]\nline 1:418: Column [lk] has conflicting data types in FORK branches: [NULL] and [KEYWORD]
```
This seems to be https://github.com/elastic/elasticsearch/issues/142543.

I think this is just missing the corresponding mute in GenerativeRestTest.

But I also see different failures:
```
java.lang.AssertionError: query: from languages_lookup,messa* | keep `language_code` | mv_expand `language_code` | eval  `language_code` = language_code + language_code, `language_code` = -92733565570034336, ginUJgbg = true, GKlZDeqJJn = 2107629793 | SAMPLE 0.29984937331915007	
error: Expecting [3] columns, got [4]
```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -431,9 +431,6 @@ tests:
 - class: org.elasticsearch.xpack.ccr.CcrTimeSeriesDataStreamsIT
   method: testCrossClusterReplicationForTSDB
   issue: https://github.com/elastic/elasticsearch/issues/142797
-- class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/142426
 - class: org.elasticsearch.xpack.esql.heap_attack.HeapAttackSubqueryIT
   method: testManyRandomKeywordFieldsInSubqueryIntermediateResultsWithSortOneField
   issue: https://github.com/elastic/elasticsearch/issues/141202
@@ -449,9 +446,6 @@ tests:
 - class: org.elasticsearch.packaging.test.DebMetadataTests
   method: test05CheckLintian
   issue: https://github.com/elastic/elasticsearch/issues/142819
-- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/142426
 - class: org.elasticsearch.xpack.search.AsyncSearchTaskTests
   method: testWithFailure
   issue: https://github.com/elastic/elasticsearch/issues/142821

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
@@ -101,6 +101,12 @@ public abstract class GenerativeRestTest extends ESRestTestCase implements Query
         "expected named expression for grouping; got ",
         "Time-series aggregations require direct use of @timestamp which was not found. If @timestamp was renamed in EVAL, "
             + "use the original @timestamp field instead.", // https://github.com/elastic/elasticsearch/issues/140607
+        "change point value \\[.*\\] must be numeric", // https://github.com/elastic/elasticsearch/issues/142858
+        // https://github.com/elastic/elasticsearch/issues/142860
+        "(Grok|Dissect) only supports KEYWORD or TEXT values, found expression \\[.*\\] type \\[NULL\\]",
+        // https://github.com/elastic/elasticsearch/issues/142543
+        "Column \\[.*\\] has conflicting data types in FORK branches: \\[NULL\\] and \\[.*\\]",
+        "Column \\[.*\\] has conflicting data types in FORK branches: \\[.*\\] and \\[NULL\\]",
 
         // Ts-command errors awaiting fixes
         "Output has changed from \\[.*\\] to \\[.*\\]" // https://github.com/elastic/elasticsearch/issues/134794
@@ -130,6 +136,14 @@ public abstract class GenerativeRestTest extends ESRestTestCase implements Query
      */
     private static final Pattern NULL_TYPE_MISMATCH_PATTERN = Pattern.compile(
         ".*first argument of \\[([^]]+)] is \\[null] so second argument must also be \\[null] but was \\[.*].*",
+        Pattern.DOTALL
+    );
+    /**
+     * Matches "... argument of [X] must be [Y], found value [unmapped_field] type [Z]" errors.
+     * This happens when an unmapped field ends up with a different data type that doesn't match the one of the function's argument(s).
+     */
+    private static final Pattern ANY_TYPE_MISMATCH_PATTERN = Pattern.compile(
+        ".*argument of \\[.*] must be \\[.*], found value \\[([^]]+)] type \\[.*].*",
         Pattern.DOTALL
     );
     /**
@@ -285,7 +299,8 @@ public abstract class GenerativeRestTest extends ESRestTestCase implements Query
                     return outputValidation;
                 }
             }
-            if (isUnmappedFieldError(outputValidation.errorMessage()) || isScalarTypeMismatchError(outputValidation.errorMessage())) {
+            if (isUnmappedFieldError(outputValidation.errorMessage(), result.query())
+                || isScalarTypeMismatchError(outputValidation.errorMessage())) {
                 return outputValidation;
             }
             if (isFirstLastSameFieldError(outputValidation.errorMessage(), result.query())) {
@@ -305,7 +320,8 @@ public abstract class GenerativeRestTest extends ESRestTestCase implements Query
                 return;
             }
         }
-        if (isUnmappedFieldError(query.exception().getMessage()) || isScalarTypeMismatchError(query.exception().getMessage())) {
+        if (isUnmappedFieldError(query.exception().getMessage(), query.query())
+            || isScalarTypeMismatchError(query.exception().getMessage())) {
             return;
         }
         if (isFirstLastSameFieldError(query.exception().getMessage(), query.query())) {
@@ -338,7 +354,10 @@ public abstract class GenerativeRestTest extends ESRestTestCase implements Query
      *   <li>"Rule execution limit [100] reached" - can happen with complex plans involving "nullify" unmapped fields</li>
      * </ul>
      */
-    private static boolean isUnmappedFieldError(String errorMessage) {
+    private static boolean isUnmappedFieldError(String errorMessage, String query) {
+        if (query.startsWith(SET_UNMAPPED_FIELDS_PREFIX) == false) {
+            return false;
+        }
         String errorWithoutLineBreaks = ERROR_MESSAGE_LINE_BREAK.matcher(errorMessage).replaceAll("");
         // Try the more specific pattern first (with suggestion)
         Matcher matcher = UNKNOWN_COLUMN_WITH_SUGGESTION_PATTERN.matcher(errorWithoutLineBreaks);
@@ -359,7 +378,12 @@ public abstract class GenerativeRestTest extends ESRestTestCase implements Query
             String expression = matcher.group(1);
             return UNMAPPED_NAMES.stream().anyMatch(expression::contains);
         }
-
+        // a non-NULL type mismatch for an unmapped field name used in a function as argument
+        matcher = ANY_TYPE_MISMATCH_PATTERN.matcher(errorWithoutLineBreaks);
+        if (matcher.matches()) {
+            String expression = matcher.group(1);
+            return UNMAPPED_NAMES.stream().anyMatch(expression::contains);
+        }
         // https://github.com/elastic/elasticsearch/issues/142390
         if (errorWithoutLineBreaks.contains("Rule execution limit [100] reached")) {
             return true;

